Split/Matchers/Semver/Semver.swift

```diff
@@ -155,6 +103,82 @@ class Semver: Equatable {
         return lhs.version == rhs.version
     }
 
+    private func getAndRemoveMetadataIfExists(_ version: String) throws -> (metadata: String?, versionWithoutMetadata: String) {
+        var vWithoutMetadata = ""
+        var tMetadata = ""
+        if let index = version.firstIndex(of: kMetadataDelimiter) {
+            let metadataIndex = version.index(after: index)
+
+            if metadataIndex >= version.endIndex {
+                throw SemverParseError.invalidVersionFormat("Unable to convert to Semver, incorrect metadata")
+            }
+
+            tMetadata = try getMetadata(version, metadataIndex)
+
+            let versionSubstring = version[..<index]
+
+            vWithoutMetadata = String(versionSubstring)
+        } else {
+            vWithoutMetadata = version
+        }
+
+        return (tMetadata, vWithoutMetadata)
+    }
+
+    private func getMetadata(_ version: String, _ index: String.Index) throws -> String {
+        let metadataString = String(version[index...])
+
+        if metadataString.isEmpty {
+            throw SemverParseError.invalidVersionFormat("Unable to convert to Semver, incorrect metadata")
+        }
+
+        return metadataString
+    }
+
+    private func getAndRemovePreReleaseIfExists(_ vWithoutMetadata: String) throws -> (preRelease: [String], isStable: Bool, versionWithoutPreRelease: String) {
+        var vWithoutPreRelease = ""
+        var tPreRelease: [String] = []
+        var tIsStable = true
+        if let index = vWithoutMetadata.firstIndex(of: kPreReleaseDelimiter) {
+            let preReleaseDataIndex = vWithoutMetadata.index(after: index)
+
+            guard preReleaseDataIndex < vWithoutMetadata.endIndex else {
+                throw SemverParseError.invalidVersionFormat("Unable to convert to Semver, incorrect pre release data")
+            }
+
+            let preReleaseData = String(vWithoutMetadata[preReleaseDataIndex...])
+            let preReleaseComponents = preReleaseData.split(separator: kValueDelimiter, omittingEmptySubsequences: false).map { String($0) }
+
+            if preReleaseComponents.isEmpty || preReleaseComponents.contains(where: { $0.isEmpty }) {
+                throw SemverParseError.invalidVersionFormat("Unable to convert to Semver, incorrect pre release data")
+            }
+
+            tPreRelease = preReleaseComponents
+            tIsStable = false
+
+            vWithoutPreRelease = String(vWithoutMetadata[..<index])
+        } else {
+            tIsStable = true
+            vWithoutPreRelease = vWithoutMetadata
+        }
+
+        return (tPreRelease, tIsStable, vWithoutPreRelease)
+    }
+
+    private func getMajorMinorAndPatch(_ vWithoutPreRelease: String) throws -> (major: Int64, minor: Int64, patch: Int64) {
```

Comment:
`func baseVersion(withoutPreRelease: String) -> SemverModel {`
Also please use named parameters. It makes the code self descriptive.